crates/node-api/src/engine/mod.rs

```diff
@@ -150,6 +160,16 @@ pub trait EngineTypes: Send + Sync {
         + Unpin;
 
     // TODO: payload type
+
+    /// Validates the presence or exclusion of fork-specific fields based on the payload attributes
+    /// and the message version.
+    fn validate_version_specific_fields<Type>(
+        chain_spec: &ChainSpec,
+        version: EngineApiMessageVersion,
+        payload_or_attrs: &PayloadOrAttributes<'_, Type>,
+    ) -> Result<(), AttributesValidationError>
+    where
+        Type: PayloadAttributes;
```

Comment:
see other comment about redundant generic here

Reply:
generic is removed now